extensions/sql-migration/src/dialog/assessmentResults/savedAssessmentDialog.ts

```diff
@@ -0,0 +1,144 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as constants from '../../constants/strings';
+import { MigrationStateModel } from '../../models/stateMachine';
+import { WizardController } from '../../wizard/wizardController';
+
+export class SavedAssessmentDialog {
+
+	private static readonly OkButtonText: string = constants.NEXT_LABEL;
+	private static readonly CancelButtonText: string = constants.CANCEL_LABEL;
+
+	private _isOpen: boolean = false;
+	private dialog: azdata.window.Dialog | undefined;
+	private _rootContainer!: azdata.FlexContainer;
+	private stateModel: MigrationStateModel;
+	private context: vscode.ExtensionContext;
+	private _disposables: vscode.Disposable[] = [];
+
+	constructor(context: vscode.ExtensionContext, stateModel: MigrationStateModel) {
+		this.stateModel = stateModel;
+		this.context = context;
+	}
+
+	private async initializeDialog(dialog: azdata.window.Dialog): Promise<void> {
+		return new Promise<void>((resolve, reject) => {
+			dialog.registerContent(async (view) => {
+				try {
+					this._rootContainer = this.initializePageContent(view);
+					await view.initializeModel(this._rootContainer);
+					this._disposables.push(dialog.okButton.onClick(async e => {
+						await this.execute();
+						this._disposables.forEach(
+							d => { try { d.dispose(); } catch { } }
+						);
```

Comment:
can the disposable cleanup be moved to the dialog -> onClosed or equivalent event.